components/files/file.jsx

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Files design pattern](https://lightningdesignsystem.com/components/files/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+import { FILES_FILE } from '../../utilities/constants';
+
+import FileFigure from './private/file-figure';
+import FileActions from './private/file-actions';
+
+const displayName = FILES_FILE;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	crop: PropTypes.oneOf(['16-by-9', '4-by-3', '1-by-1']),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 *  Action to be done on clicking download button; doesnt show download button if empty
+	 */
+	onClickDownload: PropTypes.func,
```

Comment:
Please add an `onClickImage` function to be applied to the `a` tag around the image, so that more advanced behaviors could be rigged up there instead of just a link.

Reply:
@aswinshenoy this needs to be added